tfhe/src/integer/server_key/radix/mod.rs

```diff
@@ -756,4 +756,45 @@ impl ServerKey {
             self.propagate(ctxt, i);
         }
     }
+
+    /// Reverse the bits of the unsigned integer
+    ///
+    /// # Example
+    ///
+    ///```rust
+    /// use tfhe::integer::{gen_keys_radix, IntegerCiphertext};
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// let num_blocks = 4;
+    ///
+    /// // Generate the client key and the server key:
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, num_blocks);
+    ///
+    /// let msg = 0b10110100_u8;
+    ///
+    /// let ct = cks.encrypt(msg);
+    ///
+    /// // Compute homomorphically an addition:
+    /// let mut ct_res = sks.reverse_bits_unsigned(&ct);
+    ///
+    /// // Decrypt:
+    /// let res: u8 = cks.decrypt(&ct_res);
+    /// assert_eq!(msg.reverse_bits(), res);
+    /// ```
+    pub fn reverse_bits_unsigned(&self, ct: &RadixCiphertext) -> RadixCiphertext {
+        let message_modulus = self.message_modulus().0 as u64;
+
+        let lut = self.key.generate_lookup_table(|x| {
+            (x % message_modulus).reverse_bits() >> (64 - message_modulus.ilog2())
+        });
+
+        let blocks = ct
+            .blocks
+            .iter()
+            .rev()
+            .map(|block| self.key.apply_lookup_table(block, &lut))
+            .collect();
```

Comment:
This should become reversive_bits_parallelized and take  `T where T: IntegerRadixCiphertext` as reversing bits on a signed value is possible and its the same thing

And the loop shall be `ct.blocks().par_iter()` otherwise the latency is num blocks and not 1 PBS (assuming enough threads)

Reply:
Done